crates/cast/tests/e2e/account/add.rs

```diff
@@ -0,0 +1,177 @@
+        "--salt",
+        "0x3",
+        "--class-hash",
+        "0x4",
+        "--add-profile",
+    ];
+
+    let snapbox = Command::new(cargo_bin!("sncast"))
+        .current_dir(&current_dir)
+        .args(args);
+
+    snapbox.assert().stdout_matches(indoc! {r#"
+        command: account add
+        add_profile: Profile successfully added to Scarb.toml
+    "#});
+
+    let mut file = current_dir.clone();
+    file.push(Utf8PathBuf::from(accounts_file));
+
+    let contents = fs::read_to_string(file).expect("Unable to read created file");
+    let contents_json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+    assert_eq!(
+        contents_json,
+        json!(
+            {
+                "alpha-goerli": {
+                  "my_account_add": {
+                    "address": "0x1",
+                    "class_hash": "0x4",
+                    "deployed": false,
+                    "private_key": "0x2",
+                    "public_key": "0x759ca09377679ecd535a81e83039658bf40959283187c654c5416f439403cf5",
+                    "salt": "0x3",
+                  }
+                }
+            }
+        )
+    );
+
+    let contents =
+        fs::read_to_string(current_dir.join("Scarb.toml")).expect("Unable to read Scarb.toml");
+    assert!(contents.contains("[tool.sncast.my_account_add]"));
+    assert!(contents.contains("account = \"my_account_add\""));
+
+    fs::remove_dir_all(current_dir).unwrap();
+}
+
+#[tokio::test]
+pub async fn test_detect_deployed() {
+    let accounts_file = "./tmp/accounts.json";
+    _ = fs::remove_file(accounts_file);
+
+    let args = vec![
+        "--url",
+        URL,
+        "--accounts-file",
+        accounts_file,
+        "account",
+        "add",
+        "--name",
+        "my_account_add",
+        "--address",
+        DEVNET_PREDEPLOYED_ACCOUNT_ADDRESS,
+        "--private-key",
+        "0x5",
+    ];
+
+    let snapbox = runner(&args);
+
+    snapbox.assert().stdout_matches(indoc! {r#"
+        Contract detected as deployed on chain
+        command: account add
+        add_profile: --add-profile flag was not set. No profile added to Scarb.toml
+    "#});
+
+    let contents = fs::read_to_string(accounts_file).expect("Unable to read created file");
+    let contents_json: serde_json::Value = serde_json::from_str(&contents).unwrap();
+    assert_eq!(
+        contents_json,
+        json!(
+            {
+                "alpha-goerli": {
+                  "my_account_add": {
+                    "address": DEVNET_PREDEPLOYED_ACCOUNT_ADDRESS,
+                    "deployed": true,
+                    "private_key": "0x5",
+                    "public_key": "0x788435d61046d3eec54d77d25bd194525f4fa26ebe6575536bc6f656656b74c"
+                  }
+                }
+            }
+        )
+    );
+
+    fs::remove_dir_all(Utf8PathBuf::from(accounts_file).parent().unwrap()).unwrap();
+}
```

Comment:
let's add some more tests, to for example see what happens when one of the arguments is missing/malformed